http://bugs.gentoo.org/96882

date: 2005/06/23 20:09:27;  author: eggert;  state: Exp;  lines: +29 -34
Fix some more shell quoting problems.  Prompted by a bug report
from Justace Clutter.
* lib/autoconf/general.m4 (_AC_INIT_SRCDIR): Allow special 
characters in $ac_unique_file.

--- lib/autoconf/general.m4
+++ lib/autoconf/general.m4
@@ -476,20 +476,20 @@
   # Try the directory containing this script, then its parent.
   ac_confdir=`AS_DIRNAME(["$[0]"])`
   srcdir=$ac_confdir
-  if test ! -r $srcdir/$ac_unique_file; then
+  if test ! -r "$srcdir/$ac_unique_file"; then
     srcdir=..
   fi
 else
   ac_srcdir_defaulted=no
 fi
-if test ! -r $srcdir/$ac_unique_file; then
+if test ! -r "$srcdir/$ac_unique_file"; then
   if test "$ac_srcdir_defaulted" = yes; then
     AC_MSG_ERROR([cannot find sources ($ac_unique_file) in $ac_confdir or ..])
   else
     AC_MSG_ERROR([cannot find sources ($ac_unique_file) in $srcdir])
   fi
 fi
-(cd $srcdir && test -r ./$ac_unique_file) 2>/dev/null ||
+(cd $srcdir && test -r "./$ac_unique_file") 2>/dev/null ||
   AC_MSG_ERROR([sources are in $srcdir, but `cd $srcdir' does not work])
 dnl Double slashes in pathnames in object file debugging info
 dnl mess up M-x gdb in Emacs.
